src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -241,9 +243,29 @@ public DataAccessRequest insertDraftDataAccessRequest(User user, DataAccessReque
             now,
             dar.getData()
         );
+
+        syncDataAccessRequestDatasets(dar.getData().getDatasetIds(), dar.getReferenceId());
+
         return findByReferenceId(dar.getReferenceId());
     }
 
+    /**
+     * First delete any rows with the current reference id. This will allow us to keep (referenceId, dataset_id) unique
+     * Takes in a list of datasetIds and a referenceId and adds them to the dar_dataset collection
+     *
+     * @param datasetIds List of Integers that represent the datasetIds
+     * @param referenceId ReferenceId of the corresponding DAR
+     */
+    private void syncDataAccessRequestDatasets(List<Integer> datasetIds, String referenceId) {
+        dataAccessRequestDAO.deleteDARDatasetRelationByReferenceId(referenceId);
+
+        if (!datasetIds.isEmpty()) {
+            datasetIds.forEach(id -> {
+                dataAccessRequestDAO.insertDARDatasetRelation(referenceId, id);
+            });
+        }
```

Comment:
Suggestion: Batch insert these instead of doing inserts in a loop.

1. Look at MatchDAO.insertAll for a model for how to do this
2. Create a value class that we can use as an insert

```
public class DarDataset() {
    public String referenceId;
    public Integer datasetId;
    ....
}
```

```
    @SqlBatch("INSERT INTO dar_dataset (reference_id, dataset_id) VALUES (:referenceId, :datasetId)")
    void insertAllDarDatasets(@BindBean List<DarDataset> darDatasets);
```